### DISCUSSION TITLE
Different results with the same parameter values ​​in the model(Random state)

### DISCUSSION THREAD

**Discussion by IpekInceler** [ourownstory/neural_prophet#224] [Q&A - get help using NeuralProphet] [answered]
Hi Everyone,
First of all I like to use the Library. ❤️ 
I have a problem. I'm working on a time series data. 
Although I use the same parameter values, I get different mape values ​​every time I run the model. Below is a screenshot of the parameters I use. I saw that there were no parameters such as random states or seeds that I could define in the model.  How can I solve this problem??
<img width="310" alt="Ekran Resmi 2020-12-15 15 11 33" src="https://user-images.githubusercontent.com/50381965/102213517-e7aa5780-3ee7-11eb-96ae-80e69263f9ec.png">

Best regards and thanks, 

Ipek

**Comment by ourownstory** [comment, marked as answer]
hi @IpekInceler  and thank you!

This is documented at the end of Quickstart in chapter "Reproducibility"
https://ourownstory.github.io/neural_prophet/

For more details, have a look at #147

Let us know if this does not work for you!

**Reply by IpekInceler** [reply]
Hi @ourownstory 
 I looked at the link [https://ourownstory.github.io/neural_prophet/]() and solved my questions
Thanks so much for the suggestions.

**Comment by codeananda** [comment]
This is helpful, but I don't think it covers everything. The code for [`set_random_seed`](https://github.com/ourownstory/neural_prophet/blob/0ab224a35da192944ab040fbc900c277c5fed374/neuralprophet/utils.py#L639) is just 


```python
np.random.seed(seed)
torch.manual_seed(seed)
```

This PyTorch [thread](https://github.com/pytorch/pytorch/issues/7068) discusses the many ways you can set a random seed and that the above two lines may not be enough. The [most popular answer](https://github.com/pytorch/pytorch/issues/7068#issuecomment-484918113) recommends doing this

```python
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    torch.cuda.manual_seed_all(seed)  # if you are using multi-GPU.
    np.random.seed(seed)  # Numpy module.
    random.seed(seed)  # Python random module.
    torch.manual_seed(seed)
    torch.backends.cudnn.benchmark = False
    torch.backends.cudnn.deterministic = True
```
